discovery-provider/src/tasks/index_metrics.py

```diff
@@ -126,7 +136,144 @@ def refresh_metrics_matviews(db):
         session.execute('REFRESH MATERIALIZED VIEW app_name_metrics_all_time')
         logger.info('index_metrics.py | refreshed metrics matviews')
 
-######## CELERY TASK ########
+def get_all_other_nodes():
+    shared_config = aggregate_metrics.shared_config
+    eth_web3 = aggregate_metrics.eth_web3
+    eth_registry_address = aggregate_metrics.eth_web3.toChecksumAddress(
+        shared_config["eth_contracts"]["registry"]
+    )
+    eth_registry_instance = eth_web3.eth.contract(
+        address=eth_registry_address, abi=eth_abi_values["Registry"]["abi"]
+    )
+    sp_factory_address = eth_registry_instance.functions.getContract(
+        sp_factory_registry_key
+    ).call()
+    sp_factory_inst = eth_web3.eth.contract(
+        address=sp_factory_address, abi=eth_abi_values["ServiceProviderFactory"]["abi"]
+    )
+    num_discovery_providers = sp_factory_inst.functions.getTotalServiceTypeProviders(discovery_node_service_type).call()
+    logger.info(f"number of discovery providers: {num_discovery_providers}")
+    service_infos = [sp_factory_inst.functions.getServiceEndpointInfo(discovery_node_service_type, i).call() \
+        for i in range(1, num_discovery_providers + 1)]
+
+    logger.info(f"this node's delegate owner wallet: {shared_config['delegate']['owner_wallet']}")
+    all_other_nodes = [service_info[1] for service_info in service_infos \
+        if service_info[3] != shared_config["delegate"]["owner_wallet"]]
+    logger.info(f"all the other nodes: {all_other_nodes}")
+    return all_other_nodes
+
+def get_metrics(endpoint, start_time):
+    try:
+        route_metrics_endpoint = f"{endpoint}/v1/metrics/aggregates/routes/cached?start_time={start_time}"
+        logger.info(f"route metrics request to: {route_metrics_endpoint}")
+        route_metrics_response = requests.get(route_metrics_endpoint, timeout=3)
```

Comment:
why the 3s timeout here? not that it would take longer, but i also don't see harm with this being something grossly larger like 10